misk/src/test/kotlin/misk/web/actions/UpstreamResourceInterceptorTest.kt

```diff
@@ -0,0 +1,116 @@
+package misk.web.actions
+
+import misk.web.NetworkChain
+import misk.web.Request
+import misk.web.Response
+import misk.web.ResponseBody
+import misk.web.marshal.StringResponseBody
+import okhttp3.HttpUrl
+import okhttp3.mockwebserver.MockResponse
+import okhttp3.mockwebserver.MockWebServer
+import okio.Buffer
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.jupiter.api.AfterEach
+import org.junit.jupiter.api.BeforeEach
+import org.junit.jupiter.api.Test
+import java.net.HttpURLConnection
+import kotlin.reflect.KFunction
+
+internal class UpstreamResourceInterceptorTest {
+  val upstreamServer = MockWebServer()
+  lateinit var mapping: UpstreamResourceInterceptor.Mapping
+  lateinit var interceptor: UpstreamResourceInterceptor
+
+  // TODO: probably discourage path rewriting
```

Comment:
what action do we take on this?

Reply:
Still thinking about this one, will post a full reply before PR is closed.

Reply:
The current implementation works as follows
`localhost:8080/_admin/config/js/config.js` matches on `/_admin/config` and forwards to `localhost:3001/{upstreamPrefix}/_admin/config/js/config.js`.

Since the chain URL suffix is simply added to the upstreamPrefix there is no possibility of complex rewriting of the URL suffix (which I think was the goal of the stated todo).

If we wanted to ensure that the upstreamPrefix is always empty and thus `localhost:8080/{match}/` maps directly to `localhost:3001/{match}/` then we could set a requirement that the upstream URL always has 0 path segments. Though I don't think this would likely impact the Webpack-Dev-Server use of the class, I'm not sure whether the additional requirement would cause issues in later use if we wanted to map some links to `localhost:3001/build/{match}` for example. 

For now, should we could add in the additional requirement and remove it if it becomes a blocker?

Reply:
Not sure I follow.

Reply:
My bad, take 2:

If we wanted to eliminate all URL rewriting we can just force that the upstreamBaseURL is always the `host` + `localPathPrefix`. This way, the incoming links match on the `localPathPrefix` and are simply forwarded to the right `host`, with no URL rewriting.

Reply:
Gotcha. I don’t feel strongly!